x-pack/plugins/stack_connectors/public/connector_types/gen_ai/use_get_dashboard.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { useKibana } from '@kbn/triggers-actions-ui-plugin/public';
+import { getDashboardId } from './constants';
+import { getDashboard } from './api';
+import * as i18n from './translations';
+
+interface Props {
+  connectorId: string;
+}
+
+export interface UseGetDashboard {
+  dashboardUrl: string | null;
+  isLoading: boolean;
+}
+
+export const useGetDashboard = ({ connectorId }: Props): UseGetDashboard => {
+  const {
+    dashboard,
+    http,
+    notifications: { toasts },
+    spaces,
+  } = useKibana().services;
+
+  const [spaceId, setSpaceId] = useState<string | null>(null);
+
+  useEffect(() => {
+    let didCancel = false;
+    if (spaces) {
+      spaces.getActiveSpace().then((space) => {
+        if (!didCancel) setSpaceId(space.id);
+      });
+    }
+
+    return () => {
+      didCancel = true;
+    };
+  }, [spaces]);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [dashboardUrl, setDashboardUrl] = useState<string | null>(null);
+  const [dashboardCheckComplete, setDashboardCheckComplete] = useState<boolean>(false);
+  const abortCtrl = useRef(new AbortController());
+
+  const setUrl = useCallback(
+    (dashboardId: string) => {
+      const url = dashboard?.locator?.getRedirectUrl({
+        query: {
```

Comment:
A nice enhancement for later would be looking into making this one of the "pilled" filter thingees, instead of the query.  That would leave the query empty for customers to do further querying without having to worry about the connector match clause.  Not actually sure that's possible,  but think it is.